app/jobs/regular/change_display_name.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChangeDisplayName < ::Jobs::Base
+    sidekiq_options queue: "low"
+
+    def execute(args)
+      @user = User.find_by(id: args[:user_id])
+
+      return unless user.present?
+
+      # We need to account for the case where the instance allows
+      # name to be empty by falling back to username.
+      @old_display_name = (args[:old_name].presence || user.username).unicode_normalize
+      @new_display_name = (args[:new_name].presence || user.username).unicode_normalize
+
+      @quote_rewriter = QuoteRewriter.new(user.id)
+
+      update_posts
+      update_revisions
+    end
+
+    private
+
+    attr_reader :user, :old_display_name, :new_display_name, :quote_rewriter
+
+    def update_posts
+      Post
+        .with_deleted
+        .joins(quoted("posts.id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |post| update_post(post) }
+    end
+
+    def update_revisions
+      PostRevision
+        .joins(quoted("post_revisions.post_id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |revision| update_revision(revision) }
+
+      PostRevision
+        .joins(quoted("post_revisions.post_id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |revision| update_revision(revision) }
+    end
+
+    def update_post(post)
+      post.raw = update_raw(post.raw)
+      post.cooked = update_cooked(post.cooked)
+
+      post.update_columns(raw: post.raw, cooked: post.cooked)
+
+      SearchIndexer.index(post, force: true) if post.topic
```

Comment:
Indexing for search can be an expensive operation and I'm not sure if we need to do this immediately within this job. Perhaps we can just update `PostSearchData#version` of the post to `null` and let the `ReindexSearch` job pick it up for reindexing automatically? 

https://github.com/discourse/discourse/blob/3802d0de9d3309ff2ba8e6e63fb4ce4fecc69c87/app/jobs/scheduled/reindex_search.rb#L152-L166

Reply:
I think it'll be be good to check if we're scrub quotes when indexing `Post#cooked` for search. If the quotes are being scrubbed then there is no need to reindex.

Reply:
We do not seem to. The scrubber only seems to scrub light-boxes, mentions, emoji, and certain links.

Reply:
> Is there a benefit to deferring this, though?

One benefit of deferring indexing here is to defer the risk of indexing failure to a job that is dedicated for search indexing. Search indexing does not need to happen immediately so we don't want to carry the risk of search indexing failure crashing this job.